libminifi/src/core/ProcessSession.cpp

```diff
@@ -478,15 +466,9 @@ void ProcessSession::import(std::string source, const std::shared_ptr<core::Flow
       throw Exception(FILE_OPERATION_EXCEPTION, "File Import Error");
     }
   } catch (std::exception &exception) {
-    if (flow) {
```

Comment:
Where does it happen now? 
I guess this was here for a reason.

Reply:
it previously checked if it had the claim that we just created, but an exception can only come from before the set happens so it doesn't make much sense to do so, IMO there should be two global `ResourceClaim`s, the `InvalidResourceClaim` and the `EmptyResourceClaim` so we could abandon the "empty content path signifies an empty resourceclaim" and have a non-null claim even for these "faulty" scenarios

Reply:
the behavior doesn't change (I mean I changed it, and now changed it back), that if the import fails the flowFile retains its previous contentClaim